Add setting to only show events starting in x minutes.

KaiOelfke opened the pull request:
### Status
**READY**

### Description
This PR adds a setting to only show events in x minutes. It implements the feature request #176. It's only filtering out future events that start later then the configured threshold. Current or past events are displayed as configured by other settings.

![Bildschirmfoto 2021-02-12 um 19 08 41](https://user-images.githubusercontent.com/1190948/107805499-baeab300-6d65-11eb-89d5-20d8ca377e2b.png)


### Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

1. Have the next event in e.g. 60 minutes.
2. Open Settings and configure it to show only events starting in 30 minutes.

The event should not be displayed in the title.

3. Move event to in 25 minutes.

The event should be now displayed in the title.

jgoldhammer commented on MeetingBar/Views/Preferences/AppearanceTab.swift:
```diff
@@ -75,9 +75,7 @@ struct StatusBarSection: View {
             HStack {
                 Picker("Time", selection: $eventTimeFormat) {
                     Text("show").tag(EventTimeFormat.show)
-                    if #available(OSX 11.0, *) {} else {
```
issue : Please reintegrate the change from upstream. We want to hide this option in Big Sur...

KaiOelfke replied:
That change wasn't intended. Thanks for the heads up.

jgoldhammer commented on MeetingBar/Views/Preferences/AppearanceTab.swift:
```diff
@@ -140,14 +140,18 @@ struct EventsSection: View {
                 }
             }
+            HStack {
+                Toggle("Show only events starting in", isOn: $showEventMaxTimeUntilEventEnabled)
```
**suggestion**: Can we change this to "Show only next event starting within next"? Or what do you think @leits ?

jgoldhammer replied:
@leits question: Currently this option is in the events section. Should we move it to statusbar section to make clear that this only affects the statusbar and not the menu?

KaiOelfke replied:
I agree, it makes more sense in the status bar section and "Show only next event starting within" is clearer copy.

jgoldhammer reviewed (changes requested):
Overall looks good to me. Please lets discuss my questions .

leits commented:
What worries me is that if there are no meetings at the next chosen time, the icon will be the same as when all the meetings are over. What do you think about it, @KaiOelfke, @jgoldhammer ?

jgoldhammer commented:
> What worries me is that if there are no meetings at the next chosen time, the icon will be the same as when all the meetings are over. What do you think about it, @KaiOelfke, @jgoldhammer ?

That would be indeed a problem for the user. It would mean to read all possible next event candidates from the apple calendar and show a different icon if there is another next event for today or today+tomorrow. Does this makes sense?

leits commented:
Yes, I think so.
Maybe the app icon?

KaiOelfke commented:
Maybe, I'll have time this weekend to look at this. There's also the SF Symbol: calendar.badge.clock

KaiOelfke commented:
I moved the logic to updateTitle and added an enum to handle the different cases. I'm not sure about the title logic. But the icon for the threshold case should be the app icon or `iconCalendar` and not the checkmark one.

leits commented on MeetingBar/AppDelegate.swift:
```diff
@@ -413,7 +427,7 @@ class AppDelegate: NSObject, NSApplicationDelegate, UNUserNotificationCenterDele
         }
         preferencesWindow = NSWindow(
-            contentRect: NSRect(x: 0, y: 0, width: 700, height: 550),
+            contentRect: NSRect(x: 0, y: 0, width: 700, height: 600),
```
```suggestion
            contentRect: NSRect(x: 0, y: 0, width: 700, height: 550),
```
